frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -116,10 +116,12 @@ export const triggerPipeline = (
   pipeline: Pipeline,
   latestRun: PipelineRun,
   redirectURL?: string,
+  onTrigger?: (PipelineRun: PipelineRun) => void,
 ) => {
   k8sCreate(PipelineRunModel, newPipelineRun(pipeline, latestRun))
-    .then(() => {
-      if (redirectURL) {
+    .then((res) => {
+      onTrigger && onTrigger(res);
+      if (redirectURL && !onTrigger) {
         redirectToResourceList(redirectURL);
```

Comment:
The other redirects do this `redirectToResourceList(`${redirectURL}/${res.metadata.name}`);`
Why do they append the name while this one doesn't?

Reply:
Removed the redirectURL string and updated it to have onSubmit handlers